rasa/shared/core/generator.py

```diff
@@ -217,6 +217,40 @@ def update(self, event: Event, skip_states: bool = False) -> None:
 
             self._append_current_state()
 
+    def fingerprint(self) -> Text:
```

Comment:
Can any of this functionality be extracted into the parent class `DialogueStateTracker`?

Reply:
Seems that `sender_id` and `slots` are common attributes - should I create a fingerprint of `DialogueStateTracker` using these attributes? Then how do I combine the resulting hash of `super().fingerprint()` in `TrackerWithCachedStates` with the its other attributes (`domain`, `_states_for_hashing`)?

Reply:
Well now we don't need the `_states_for_hashing` can we just move this whole implementation to the superclass?

Reply:
`DialogueStateTracker` doesn't have `self.domain` as attribute. Let me know how best to proceed?

Reply:
True, good catch. 
So the normal way to handle this would be to create `DialogueStateTracker.fingerprint` that fingerprints everything apart from the domain, and then `TrackerWithCachedStates.fingerprint` would add/combine the domain to `super().fingerprint()`. e.g.

```
get_dictionary_fingerprint({"tracker": super().fingerprint(), "domain": self.domain})
```

However, I we may not even need the domain, as I _think_ it's just be used for creating the states. In which case you could just implement it on `DialogueStateTracker.fingerprint`.
@wochinge wdyt?

Reply:
Yikes, it seems `mypy` doesn't allow adding an object to the dictionary, I got this CI error:
```
Dict entry 0 has incompatible type "str": "Domain"; expected "str": "str"  [dict-item]
```
I'll wait for the final decision on whether to move this method to the parent class before making any changes.

Reply:
> However, I we may not even need the domain, as I think it's just be used for creating the states. In which case you could just implement it on DialogueStateTracker.fingerprint.

I agree. As soon as the parent node which provides the domain has a changed output this will lead to a "fingerprint miss" anyway

Reply:
Ok, will remove the domain and move the implementation to the parent class.